shell/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -185,41 +193,48 @@ class IPCRenderer : public gin::Wrappable<IPCRenderer> {
 
     // A task is posted to a worker thread to execute the request so that
     // this thread may block on a waitable event. It is safe to pass raw
-    // pointers to |result| and |response_received_event| as this stack frame
-    // will survive until the request is complete.
+    // pointers to |result| and |pending_sync_response_event_| as this stack
+    // frame will survive until the request is complete.
 
-    base::WaitableEvent response_received_event;
     task_runner_->PostTask(
-        FROM_HERE, base::BindOnce(&IPCRenderer::SendMessageSyncOnWorkerThread,
-                                  base::Unretained(this),
-                                  base::Unretained(&response_received_event),
-                                  base::Unretained(&result), internal, channel,
-                                  std::move(message)));
-    response_received_event.Wait();
+        FROM_HERE,
+        base::BindOnce(&IPCRenderer::SendMessageSyncOnWorkerThread,
+                       base::Unretained(this), base::Unretained(&result),
+                       internal, channel, std::move(message)));
+    pending_sync_response_event_.Wait();
+    if (!mojo_connection_error_msg_.empty()) {
+      LOG(INFO) << "Mojo connection interuppted, likely due to the Mojo "
+                   "receiver process crashing.";
+      thrower.ThrowError(mojo_connection_error_msg_);
+    }
+    pending_sync_response_event_.Reset();
     return result;
   }
 
-  void SendMessageSyncOnWorkerThread(base::WaitableEvent* event,
-                                     blink::CloneableMessage* result,
+  void SendMessageSyncOnWorkerThread(blink::CloneableMessage* result,
                                      bool internal,
                                      const std::string& channel,
                                      blink::CloneableMessage arguments) {
-    electron_browser_ptr_->get()->MessageSync(
+    electron_browser_ptr_->MessageSync(
         internal, channel, std::move(arguments),
         base::BindOnce(&IPCRenderer::ReturnSyncResponseToMainThread,
-                       base::Unretained(event), base::Unretained(result)));
+                       base::Unretained(this), base::Unretained(result)));
   }
 
-  static void ReturnSyncResponseToMainThread(base::WaitableEvent* event,
-                                             blink::CloneableMessage* result,
-                                             blink::CloneableMessage response) {
+  void ReturnSyncResponseToMainThread(blink::CloneableMessage* result,
+                                      blink::CloneableMessage response) {
     *result = std::move(response);
-    event->Signal();
+    pending_sync_response_event_.Signal();
+  }
+  void HandleMojoConnectionError() {
+    mojo_connection_error_msg_ = "IPC connection fatally interrupted.";
```

Comment:
nit: since this is just a bool really indicating whether the connection was interrupted,
```suggestion
    mojo_connection_encountered_error_ = true;
```

and pass the string to `ThrowError` directly.